### PR TITLE
[ELY-1599] support of username* into Digest HTTP mechanism

### PR DESCRIPTION
**Pull request by hkalina**
https://issues.jboss.org/browse/ELY-1599

### CODE REVIEW

File: src/main/java/org/wildfly/security/http/impl/DigestAuthenticationMechanism.java [hunk 1]
```diff
@@ -413,4 +428,11 @@ private void fail() throws AuthenticationMechanismException {
             throw httpDigest.mechCallbackHandlerFailedForUnknownReason(t);
         }
     }
+
+    private static String decodeRfc2231(String encoded) throws UnsupportedEncodingException {
```
Comment (mchoma) [inline src/main/java/org/wildfly/security/http/impl/DigestAuthenticationMechanism.java]: Just idea. Isn't this good candidate for pulling to common place, e.g. wildfly-common?

**Comment by mchoma** [comment]
qa ack not rfe

File: src/main/java/org/wildfly/security/http/impl/DigestAuthenticationMechanism.java [hunk 2]
```diff
@@ -155,7 +158,19 @@ private void validateResponse(HashMap<String, byte[]> responseTokens, final Http
         byte[] salt = messageRealm.getBytes(UTF_8);
         boolean nonceValid = nonceManager.useNonce(nonce, salt, nonceCount);
 
-        String username = convertToken(USERNAME, responseTokens.get(USERNAME));
+        String username;
```
Comment (mchoma) [inline src/main/java/org/wildfly/security/http/impl/DigestAuthenticationMechanism.java]: Just idea. As this star thing could be in theory applied to any attribute. Could this be refactored to util method with attribute name parameter

File: src/main/java/org/wildfly/security/http/impl/DigestAuthenticationMechanism.java (hunk 1 above)
Comment (hkalina) [reply src/main/java/org/wildfly/security/http/impl/DigestAuthenticationMechanism.java]: as I dont see more places where such encoding would be used and there is no relation to existing wildfy-common utils, I would prefer to keep it here and private

**Pull request merged**